example/spec/parallel/dbRecords/DeadLetterArchiveProcessingSpec.js

```diff
@@ -0,0 +1,177 @@
+        filename: './data/collections/s3_MOD09GQ_006/s3_MOD09GQ_006.json',
+        postfix: testId,
+      });
+      response = await createCollection({ prefix: stackName, collection: testCollection });
+      if (response.statusCode !== 200) {
+        throw new Error(`createCollection failed: ${JSON.stringify(response)}`);
+      }
+
+      testProvider = {
+        id: `s3_provider_${testId}`,
+        host: 'cumulus-sandbox-fake-s3-provider',
+        protocol: 's3',
+        globalConnectionLimit: 1000,
+      };
+      response = await createProvider({ prefix: stackName, provider: testProvider });
+      if (response.statusCode !== 200) {
+        throw new Error(`createProvider failed: ${JSON.stringify(response)}`);
+      }
+
+      const testFiles = [
+        fakeFileFactory({ bucket: systemBucket }),
+        fakeFileFactory({ bucket: systemBucket }),
+      ];
+
+      testGranule = fakeGranuleFactoryV2({
+        granuleId: `MOD09GQ.${randomString()}.hdf`,
+        collectionId: constructCollectionId(testCollection.name, testCollection.version),
+        files: testFiles,
+      });
+
+      testRule = await createOneTimeRule(
+        stackName,
+        {
+          workflow: 'HelloWorldWorkflow',
+          collection: pick(testCollection, ['name', 'version']),
+          provider: testProvider.id,
+          payload: { testId },
+        }
+      );
+
+      const executionArn = await findExecutionArn(
+        stackName,
+        (execution) =>
+          get(execution, 'originalPayload.testId') === testId,
+        { timestamp__from: ingestTime },
+        { timeout: 60 }
+      );
+
+      const cumulusMessage = fakeCumulusMessageFactory({
+        cumulus_meta: {
+          state_machine: getStateMachineArnFromExecutionArn(executionArn),
+          execution_name: executionArn.split(':').pop(),
+        },
+        meta: {
+          provider: testProvider,
+          collection: {
+            name: testCollection.name,
+            version: testCollection.version,
+          },
+        },
+        payload: {
+          granules: [testGranule],
+        },
+      });
+
+      archivePath = `${stackName}/dead-letter-archive-${testId}/sqs`;
+      messageKey = `${archivePath}/${cumulusMessage.cumulus_meta.execution_name}`;
+      await putJsonS3Object(systemBucket, messageKey, cumulusMessage);
+
+      response = await postDeadLetters({
+        prefix: stackName,
+        payload: {
+          bucket: systemBucket,
+          path: archivePath,
+        },
+      });
+      if (response.statusCode !== 202) {
+        throw new Error(`postDeadLetters failed: ${JSON.stringify(response)}`);
+      }
+    } catch (error) {
+      beforeAllFailed = true;
+      console.log('beforeAll() failed, error:', error);
+    }
+  });
+
+  afterAll(async () => {
+    const ruleName = get(testRule, 'name');
+    if (ruleName) await deleteRule({ prefix: stackName, ruleName });
+    if (testCollection) {
```

Comment:
Shouldn't this fail if we wrote a granule?

Reply:
🤔 Probably.. I guess I should add a granule delete here too.

Reply:
https://github.com/nasa/cumulus/pull/2181/commits/89a057d0ac961b8e73adb5b153b9beed1da9759e